net/test/embedded_test_server/embedded_test_server.cc

```diff
@@ -824,7 +824,11 @@ void EmbeddedTestServer::ServeFilesFromDirectory(
 void EmbeddedTestServer::ServeFilesFromSourceDirectory(
     base::StringPiece relative) {
   base::FilePath test_data_dir;
+#if defined(STARBOARD)
+  CHECK(base::PathService::Get(base::DIR_TEST_DATA, &test_data_dir));
+#else
   CHECK(base::PathService::Get(base::DIR_SOURCE_ROOT, &test_data_dir));
```

Comment:
Can we instead modify base/base_paths_starboard.cc to implement support for DIR_SOURCE_ROOT and return the same as DIR_TEST_DATA?

Reply:
I don't think we can directly do this replacement. ``DIR_TEST_DATA`` has one extra ``/test`` directory inside, and not all DIR_SOURCE_ROOT should be replaced by this DIR_TEST_DATA. If you search for ``base::DIR_SOURCE_ROOT``, you will see it's still live widely in today's code base. It's a little bit strange that some places need this and some places don't. Maybe we can clean it up a little bit? But here I'm simply copy the code from trunk, so create b/330184432 and leave it in Backlog.